base/initdefs.jl

```diff
@@ -18,15 +18,75 @@ An array of the command line arguments passed to Julia, as strings.
 """
 const ARGS = String[]
 
+"""
+    ENV
+
+A dictionary that is populated from the initial process environment and whose contents
+are used to populate the environments of spawned subprocesses.
+
+!!! note
+    As of Julia 1.6, this is a plain `Dict` rather than a special type. When you modify
+    it, the process environment is not modified directly, rather when a child process
+    is started, the value of the `ENV` dict is used to initialize the environment of
+    the child process. This means that `ENV` cannot be used to communicate via the
+    current proccess's environment with libraries.
+"""
+const ENV = Dict{String,String}()
```

Comment:
Don't you need a lock?

Reply:
If it were possible to lock on an arbitrary object, then I'd say it's the callers responsibility to do the locking, but that's not possible so I guess we either need an official `ENV_LOCK` object or need to make this a threadsafe dict type, which I guess would therefore need to live in Base and might as well be exported.

Reply:
A thread-safe Dict would be useful for many use-cases in addition to this one.

Reply:
Also, regarding being able to lock on arbitrary objects, this PR doesn't implement that directly but implements a struct that bundles objects and their locks together: https://github.com/JuliaLang/julia/pull/34400

Reply:
> If it were possible to lock on an arbitrary object, then I'd say it's the callers responsibility to do the locking,

All the callers must agree on one single lock.  So, not providing it as the part of API pretty much makes it impossible to use a lock with it.